lib/Frontend/nvqpp/ConvertExpr.cpp

```diff
@@ -403,9 +413,23 @@ FunctionType QuakeBridgeVisitor::peelPointerFromFunction(Type ty) {
 }
 
 bool QuakeBridgeVisitor::VisitArraySubscriptExpr(clang::ArraySubscriptExpr *x) {
-  // TODO: add array support
-  reportClangError(x, mangler, "arrays in kernels");
-  return false;
+  auto loc = toLocation(x->getSourceRange());
+  auto rhs = popValue();
+  auto lhs = popValue();
+  Type eleTy = [&]() {
+    // NB: Check both arguments as expression may be inverted.
+    if (auto ptrTy = dyn_cast<cc::PointerType>(lhs.getType()))
+      return ptrTy.getElementType();
+    return cast<cc::PointerType>(rhs.getType()).getElementType();
+  }();
+  Type arrEleTy = [&]() {
+    // FIXME: The following dyn_cast should never fail.
```

Comment:
What exactly is the FIXME here? To add an assert? It seems like it already has a fallback path to handle the case where the dyn_cast does fail.

Reply:
This should just be a `cast` and the failing path should be deleted.

We can't do that yet because of other bugs.